x-pack/plugins/fleet/server/services/agent_policies/agent_policy_agent_count.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { AGENTS_INDEX, ES_SEARCH_LIMIT } from '../../../common/constants';
+
+/**
+ * Given a list of Agent Policy IDs, an object will be returned with the agent policy id as the key
+ * and the number of active agents using that agent policy.
+ * @param esClient
+ * @param agentPolicyIds
+ */
+export const getAgentCountForAgentPolicies = async (
+  esClient: ElasticsearchClient,
+  agentPolicyIds: string[]
+): Promise<Record<string, number>> => {
+  if (agentPolicyIds.length === 0) {
+    return {};
+  }
+
+  const searchPromise = esClient.search<
+    unknown,
+    Record<'agent_counts', { buckets: Array<{ key: string; doc_count: number }> }>
+  >({
+    index: AGENTS_INDEX,
+    body: {
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                active: 'true',
+              },
+            },
+            {
+              terms: {
+                policy_id: agentPolicyIds,
+              },
+            },
+          ],
+        },
+      },
+      aggs: {
+        agent_counts: {
+          terms: {
+            field: 'policy_id',
+            size: ES_SEARCH_LIMIT,
```

Comment:
nit: we could probably use `agentPolicyIds.length` instead, but don't think it makes much of a difference.

`ES_SEARCH_LIMIT` is 10k which I think only applies to search hits, the bucket size limit is somewhere around 65k.

Reply:
Oh wow. Ok. Yeah, then maybe just using `agentPolicyIds.length` is ok here. I will change it.